xbmc/windowing/gbm/DRMAtomic.cpp

```diff
@@ -58,14 +66,30 @@ void CDRMAtomic::DrmAtomicCommit(int fb_id, int flags, bool rendered, bool video
 
     if (modeset)
     {
+      drmModeFBPtr r;
+      r=drmModeGetFB(m_fd,fb_id);
+      if(r){
+       CLog::Log(LOGINFO, "SAMEER --> FB Details:\n");
+       CLog::Log(LOGINFO, "SAMEER:[%s:%d]  FB - [ID-%d] [W-%d x H-%d] [Pitch-%d][BPP- %d] [Depth-%d]\n", __func__, __LINE__,r->fb_id,r->width,r->height,r->pitch,r->bpp,r->depth);
+      }
+      CLog::Log(LOGINFO, "SAMEER: [%s:%d] WxH --> %d x %d\n",__func__, __LINE__, m_width,m_height);
+
       AddProperty(m_gui_plane, "SRC_X", 0);
       AddProperty(m_gui_plane, "SRC_Y", 0);
       AddProperty(m_gui_plane, "SRC_W", m_width << 16);
       AddProperty(m_gui_plane, "SRC_H", m_height << 16);
       AddProperty(m_gui_plane, "CRTC_X", 0);
       AddProperty(m_gui_plane, "CRTC_Y", 0);
-      AddProperty(m_gui_plane, "CRTC_W", m_mode->hdisplay);
-      AddProperty(m_gui_plane, "CRTC_H", m_mode->vdisplay);
+	uint32_t scale_factor = getScaling_factor(m_mode->hdisplay, m_mode->vdisplay, m_width, m_height);
+	CLog::Log(LOGINFO, "SAMEER --> [%s:%d] IS-Factor is :%ld\n", __func__, __LINE__,scale_factor );
+	AddProperty(m_gui_plane, "CRTC_W", (scale_factor*m_width));
+	AddProperty(m_gui_plane, "CRTC_H", (scale_factor*m_height));
+	uint32_t property_id = this->GetPropertyId(m_gui_plane, "SCALING_FILTER");
+	CLog::Log(LOGINFO, "SAMEER --> [%s:%d] Property_ID:%ld - SCALING_FILTER\n", __func__, __LINE__,property_id );
+	AddProperty(m_gui_plane, "SCALING_FILTER", property_id ? 1 : 0);
+	property_id = this->GetPropertyId(m_video_plane, "SCALING_FILTER");
+	CLog::Log(LOGINFO, "SAMEER --> [%s:%d] Property_ID:%ld - SCALING_FILTER\n", __func__, __LINE__,property_id );
+	AddProperty(m_video_plane, "SCALING_FILTER", property_id ? 1 : 0);
```

Comment:
This hardcodes the `SCALING_FILTER` to 1 if the property is advertised.

- What does this value mean?
- If the property doesn't exist, shouldn't we **not** set it?
- Is an hardcoded value enough to demonstrate that the new `SCALING_FILTER` property works?

Reply:
Furthermore, using hardcoded values 0 and 1 is incorrect.

Userspace must query the kernel what values the strings `"Default"` and `"Nearest Neighbor"` map to, and then use those values. Userspace *must not* assume the values can be hardcoded.

Reply:
Thanks for your comments, I will rework on those. I had plan to add enums to be more meaning.
I will consider your input and work on that.

Reply:
@emersion/ppaalanen,

From kernel I can query only current filter value (default) and it does not give me complete supported list of filters. Is it fine to define our own enum inline with kernel enum?
Does it make sense ?

enum drm_scaling_filter {
   DRM_SCALING_FILTER_DEFAULT,
    DRM_SCALING_FILTER_NEAREST_NEIGHBOR,
};
.
.
.
if(SupportsProperty(m_video_plane, "SCALING_FILTER")){
	AddProperty(m_video_plane, "SCALING_FILTER", DRM_SCALING_FILTER_NEAREST_NEIGHBOR);
}


Refer Kernel patch series @: https://patchwork.freedesktop.org/series/73883/#rev6


-Sameer Lattannavar

Reply:
No, under the current kernel UAPI rules, it is not fine.

You can query the kernel for the whole list though: `drmModeObjectGetProperties()` gives you everything you need to know.

Reply:
I imagine something like this would be what you are looking for? https://github.com/lrusak/xbmc/commit/9eac28915d8a86b9c42bd9fac2c43e555e222931

Reply:
Yes, that has the right idea.

Reply:
Thanks @ppaalanen & @lrusak for your inputs. Please suggest me if these changes are fine?
 ----------------------------------------------------------------------------------------------------
if(CServiceBroker::GetWinSystem()->UseScalingFilter()){
     if(SupportsProperty(m_gui_plane, "SCALING_FILTER")){
		uint32_t prop_id = this->GetPropertyId(m_gui_plane, "SCALING_FILTER");
		drmModePropertyPtr m_gui_prop = drmModeGetProperty(m_fd, prop_id);
		for(int i=0; i<m_gui_prop->count_enums; i++){
				if(!strcmp(m_gui_prop->enums[i].name, "Nearest Neighbor"))
						AddProperty(m_gui_plane, "SCALING_FILTER", m_gui_prop->enums[i].value);
		}
		}
        if(SupportsProperty(m_video_plane, "SCALING_FILTER")){
			uint32_t prop_id = this->GetPropertyId(m_video_plane, "SCALING_FILTER");
			drmModePropertyPtr m_video_prop = drmModeGetProperty(m_fd,prop_id);
			for(int i=0; i<m_video_prop->count_enums; i++){
					if(!strcmp(m_video_prop->enums[i].name, "Nearest Neighbor"))
							AddProperty(m_video_plane, "SCALING_FILTER", m_video_prop->enums[i].value);
			}
}
-Sameer Lattannavar